comtypes/__init__.py

```diff
@@ -488,7 +488,7 @@ def _make_dispmethods(self, methods):
                 if is_prop:
                     self.__map_case__[name[5:].lower()] = name[5:]
 
-        for (name, nargs), methods in properties.items():
+        for (name, nargs), methods in list(properties.items()):
```

Comment:
If I am not mistaken the list conversion can be omitted in an iterating context. AFAIK it is even faster that way.

Thanks for porting this to python3.

Reply:
Well, I didn't really do a proper port.. I just did a PR of what distutils output for python 3 code. Easiest and quickest way to get it back up and running. I figured changes like the one above could be handled after this PR got merged, if it even gets merged.

The distutils 2 to 3 conversion does "literal" conversions to keep things as they would be in python 2.  `dict.items()` in python 2 returned a list ad not a generator so a proper conversion would be to wrap the call in `list()`. There are going to be changes that can be made in order to make the code more efficient and also to remove blocks of code that no longer need to be there because they are there only for python 2. 

For the sake of just getting comtypes working again I think it best to leave this PR as is and make changes afterwards and have a PR for each change. I suggest this because we know that the python 2 to 3 conversion does work without breaking anything and it's best no to introduce any changes that might break something. 

While I do know that the code change you have brought up wouldn't break anything it feel it is best to wait to make those changes.

Any thought on agreement or disagreement are  helpful  and wanted.  So anyone that want to add to this conversation please chime in. 

While I am not new  to using comtypes I have not spent the time to really go over the library to understand the mechanics of it. I have put in some hours reading the code and it is quite hard to follow because of what I  call voodoo magic code. This type of code causes and IDE to not be able to determine a data path through the code which makes it even harder to understand what is happening. In the hours I have spent reading over the code I have found several places similiar to the one you have pointed out above that can be changed to increase speed.

distutils also does not fix the `__cmp__` magic method and does not create a function  for `cmp()` which will cause a failure in the code. `__gt__`, `__lt__`, `__ge__`, `__le__`, `__eq__` and `__ne__` methods need to be added in place of `__cmp__` and the proper operators need to be used instead of using `cmp()`.

Reply:
@kdschlosser That totally makes sense and I agree. 

@cmin764 I hope this gets merged and released soon

I would be happy to help with a PR to implement the changes you mentioned.

Reply:
Yes, I can help you with getting it merged, then following a release.